File: cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

struct Change
{
    int quarters;
    int dimes;
    int nickels;
    int pennies;
};


struct Change computeChange(float total)
{  
    struct Change change;
    
    change.quarters = 0;
    change.dimes = 0;
    change.nickels = 0;
    change.pennies = 0;

    int remainingChange = round(total * 100);
   
    if( remainingChange >= 25)
    {
        change.quarters = remainingChange/25;
        remainingChange -= (change.quarters * 25);
    }
    if( remainingChange >= 10)
    {
         change.dimes = remainingChange/10;
         remainingChange -= change.dimes * 10;
    
    }
    if( remainingChange >= 5)
    {
          change.nickels = remainingChange/5;
          remainingChange -= (change.nickels * 5);
    }
    if( remainingChange >= 0)
    {
        change.pennies = remainingChange/1;
        remainingChange -= (change.pennies * 1);
    }

    return change;
}


int main()
{
    float total;
    do
    {
        total = get_float("\nPlease enter an amount of money: \n"); 
    }
    while (total < 0);
        
    struct Change change = computeChange(total);

    printf("TOTAL VALUE ENTERED: %6.2f",total);
    printf("\n%i quarters (0.25)\n",change.quarters); 
    printf("\n%i dimes (0.10)\n",change.dimes); 
    printf("\n%i nickles (0.05)\n",change.nickels);
    printf("\n%i pennies (0.01)\n",change.pennies);


    printf("Change owed: %.2f\n",total); 
    printf("%i\n",change.quarters + change.dimes + change.nickels + change.pennies); 

   return 0;
}

File: cash/cash.py

class Change:
    def __init__(self, total):
        self.total = total
        self.quarters = 0
        self.dimes = 0
        self.nickels = 0
        self.pennies = 0
        self.remainingChange = round(self.total * 100)

    def computeChange(self):

        self.quarters = 0
        self.dimes = 0
        self.nickels = 0
        self.pennies = 0

        if self.remainingChange >= 25:
            self.quarters = int(self.remainingChange/25)
            self.remainingChange -= (self.quarters * 25)

        if self.remainingChange >= 10:
            self.dimes = int(self.remainingChange/10)
            self.remainingChange -= self.dimes * 10
        
        if self.remainingChange >= 5:
            self.nickels = int(self.remainingChange/5)
            self.remainingChange -= (self.nickels * 5)
        
        if self.remainingChange >= 0:
            self.pennies = int(self.remainingChange/1)
            self.remainingChange -= (self.pennies * 1)

def getUserInput():
    return float(input('total: '))

def main():

    total = getUserInput()
    while (total < 0):
        total = getUserInput()

    change = Change(total)
    change.computeChange()
    # print("TOTAL VALUE ENTERED: %6.2f", change.total)
    print(f"quarters (0.25) {change.quarters}")
    print(f"dimes (0.10) {change.dimes}")
    print(f"nickles (0.05) {change.nickels}")
    print(f"pennies (0.01) {change.pennies}")


    print("Change owed: ", change.total)
    print("number of coins: ", change.quarters + change.dimes + change.nickels + change.pennies)

main()

